mne_bids/dig.py

```diff
@@ -527,10 +497,26 @@ def _read_dig_bids(electrodes_fpath, coordsystem_fpath,
              .format(electrodes_fpath, BIDS_COORDINATE_UNITS))
         coord_frame = None
 
-    # only set montage if coordinate frame was properly parsed
+    # montage is interpretable only if coordinate frame was properly parsed
     if coord_frame is not None:
-        # read in electrode coordinates and attach to raw
-        raw = _handle_electrodes_reading(electrodes_fpath, coord_frame,
-                                         coord_unit, raw, verbose)
-
-    return raw
+        # read in electrode coordinates as a DigMontage object
+        montage = _handle_electrodes_reading(electrodes_fpath, coord_frame,
+                                             coord_unit, verbose)
+    else:
+        montage = None
+
+    if montage is not None:
+        # determine if there are problematic channels
+        ch_pos = montage._get_ch_pos()
+        nan_chs = []
+        for ch_name, ch_coord in ch_pos.items():
+            if any(np.isnan(ch_coord)) and ch_name not in raw.info['bads']:
+                nan_chs.append(ch_name)
+        if len(nan_chs) > 0:
+            warn("There are channels without locations "
+                 "(n/a) that are not marked as bad: {}".format(nan_chs))
+
+    # add montage to Raw object
+    raw.set_montage(montage, on_missing='warn', verbose=verbose)
```

Comment:
either you need to add an "if clause" (if raw is available ...), or you need to mention in the docstr that raw is REQUIRED ... currently it says OPTIONAL